test/com/esotericsoftware/kryo/SerializationCompatTest.java

```diff
@@ -109,9 +109,24 @@ public void testDefaultSerializers() throws Exception {
                 EXPECTED_DEFAULT_SERIALIZER_COUNT, defaultSerializers.size());
     }
 
-    public void testStandard () throws Exception {
+    public void testStandardOptGenerics () throws Exception {
+   	  kryo.getFieldSerializerConfig().setOptimizedGenerics(true);
         runTests(
-                "standard",
+                "standard-opt-generics",
+                new Function1<File, Input>() {
+                    public Input apply(File file) throws FileNotFoundException {
+                        return new Input(new FileInputStream(file));
+                    }
+                },
+                new Function1<File, Output>() {
+                    public Output apply(File file) throws Exception {
+                        return new Output(new FileOutputStream(file));
+                    }
+                }
+        );
+        kryo.getFieldSerializerConfig().setOptimizedGenerics(false);
+        runTests(
+                "standard-nonopt-generics",
```

Comment:
The new files (`test/resources/*-standard-nonopt-generics.ser`) created when running this `SerializationCompatTest` should be added to git.